pkg/operator/controller/dns_status.go

```diff
@@ -34,8 +44,27 @@ func (r *reconciler) syncDNSStatus(dns *operatorv1.DNS, clusterIP, clusterDomain
 	return nil
 }
 
+// conditionsUpdatedWithinLast computes if the last update for progressing and degraded conditions
+// occurred within the last given time.
+func conditionsUpdatedWithinLast(conditions []operatorv1.OperatorCondition, duration time.Duration) bool {
+	progressingUpdated := false
+	degradedUpdated := false
+	for _, c := range conditions {
+		if c.Type == operatorv1.OperatorStatusTypeProgressing {
+			if c.Status == operatorv1.ConditionFalse && c.LastTransitionTime.Add(duration).After(time.Now()) {
+				progressingUpdated = true
+			}
+		} else if c.Type == operatorv1.OperatorStatusTypeDegraded {
+			if c.Status == operatorv1.ConditionFalse && c.LastTransitionTime.Add(duration).After(time.Now()) {
+				degradedUpdated = true
+			}
+		}
+	}
+	return progressingUpdated && degradedUpdated
+}
+
 // computeDNSStatusConditions computes dns status conditions based on
-// the status of ds and clusterIP.
+// the status of dns and clusterIP.
```

Comment:
`ds` refers to the DNS daemonset.
```suggestion
// the status of the dns and node-resolver daemonsets and clusterIP.
```